File: awk/Java/awkLexerBase.java
import org.antlr.v4.runtime.CharStream;
import org.antlr.v4.runtime.Lexer;
import org.antlr.v4.runtime.Token;

public abstract class awkLexerBase extends Lexer
{
    private boolean _afterExpr = false;

    public awkLexerBase(CharStream input)
    {
        super(input);
    }

    @Override
    public Token nextToken()
    {
        Token token = super.nextToken();
        if (token.getChannel() == Token.DEFAULT_CHANNEL)
        {
            this._afterExpr = token.getType() == awkLexer.WORD
                     || token.getType() == awkLexer.NUMBER
                     || token.getType() == awkLexer.STRING
                     || token.getType() == awkLexer.BUILTIN_FUNC_NAME
                     || token.getType() == awkLexer.INCR
                     || token.getType() == awkLexer.DECR
                     || token.getType() == awkLexer.Rp
                     || token.getType() == awkLexer.Rb;
        }
        return token;
    }

    public boolean IsNotAfterExpr()
    {
        return !this._afterExpr;
    }
}

File: awk/CSharp/awkLexerBase.cs
using System;
using System.Collections.Generic;
using System.IO;
using Antlr4.Runtime;

public abstract class awkLexerBase : Lexer
{
    bool _afterExpr = false;

    public override Antlr4.Runtime.IToken NextToken()
    {
        var token = base.NextToken();
        if (token.Channel == 0) // DEFAULT_CHANNEL
        {
            this._afterExpr = token.Type == awkLexer.WORD
                     || token.Type == awkLexer.NUMBER
                     || token.Type == awkLexer.STRING
                     || token.Type == awkLexer.BUILTIN_FUNC_NAME
                     || token.Type == awkLexer.INCR
                     || token.Type == awkLexer.DECR
                     || token.Type == awkLexer.Rp
                     || token.Type == awkLexer.Rb;
        }
        return token;
    }

    public bool IsNotAfterExpr()
    {
	return ! this._afterExpr;
    }

    public awkLexerBase(ICharStream input)
        : base(input)
    {
    }

    public awkLexerBase(ICharStream input, TextWriter output, TextWriter errorOutput)
        : base(input, output, errorOutput)
    {
    }
}

